src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4609,54 +4557,78 @@ internal static ModuleLoggingGroupPolicyStatus GetModuleLoggingInformation(out I
-            if (!IsRooted(name))
-            {
-                // The manifest file should only be related to moduleBase path.
-                resolvedName = ResolveRootedFilePath(Path.Combine(moduleBase, name), this.Context);
-            }
-            else
-            {
-                resolvedName = ResolveRootedFilePath(name, this.Context);
-            }
+            pathIsResolved = false;
+            string originalName = name;
+            string originalExt = Path.GetExtension(name);
 
-            if (string.IsNullOrEmpty(resolvedName))
+            if (string.IsNullOrEmpty(originalExt))
             {
-                resolvedName = Path.Combine(moduleBase, name);
+                name += extension;
             }
-            // Again resolve the file path
-            // This is so that any relative path references are expanded to give the absolute path
-            // C:\Windows\System32\WindowsPowerShell\V1.0\Modules\Microsoft.PowerShell.WSMAN\..\..\WSMan.format.ps1xml is expanded to
-            // C:\Windows\System32\WindowsPowerShell\V1.0\WSMan.format.ps1xml
-            string resolvedName2 = ResolveRootedFilePath(resolvedName, this.Context);
-            string ext = Path.GetExtension(name);
-            string result = !string.IsNullOrEmpty(resolvedName2) ? resolvedName2 : resolvedName;
-            if (string.IsNullOrEmpty(ext))
+
+            // Try to get the resolved fully qualified path to the file.
+            // Note that, the 'IsRooted' method also returns true for relative paths, in which case we need to check for 'combinedPath' as well.
+            //  * For example, the 'Microsoft.WSMan.Management.psd1' in Windows PowerShell defines 'FormatsToProcess="..\..\WSMan.format.ps1xml"'.
+            //  * For such a module, we will have the following input when reaching this method:
+            //     - moduleBase = 'C:\Windows\System32\WindowsPowerShell\v1.0\Modules\Microsoft.WSMan.Management'
+            //     - name = '..\..\WSMan.format.ps1xml'
+            //    Check for combinedPath in this case will get us the normalized rooted path 'C:\Windows\System32\WindowsPowerShell\v1.0\WSMan.format.ps1xml'.
+            // The 'Microsoft.WSMan.Management' module in PowerShell Core was updated to not use the relative path for 'FormatsToProcess' entry,
+            // but it's safer to keep the original behavior to avoid unexpected breaking changes.
+            string combinedPath = Path.Combine(moduleBase, name);
+            string resolvedPath = IsRooted(name)
+                ? ResolveRootedFilePath(name, Context) ?? ResolveRootedFilePath(combinedPath, Context)
+                : ResolveRootedFilePath(combinedPath, Context);
+
+            // Return the path if successfully resolved.
+            if (resolvedPath != null)
             {
-                result += extension;
+                if (isImportingModule && resolvedPath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
```

Comment:
I wonder if we should name this helper function 'FixupFileNameAndLoadAssemblyAsNeeded'?  Maybe not, this comment is probably sufficient.

Reply:
Added a `<remarks>` section to mention this side effect behavior.